app/controllers/posts_controller.rb

```diff
@@ -1,7 +1,7 @@
 class PostsController < ApplicationController
   def index
+    @user = User.find_by(id: params[:user_id])
     @posts = Post.includes(:author).where(author_id: params[:user_id])
-    @user = User.includes(:posts).find(params[:user_id])
   end
 
   def show
```

Comment:
- Great job so far @citec-47, @touredri and @mercymugambi There are just a few changes I will suggest you make and you'll be on your way to the next milestone :wink: 
- [ ] Per the project requirements, you had to :

> **Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.**

- [ ] The `N+1` problem occurs when an application makes `N+1` database queries, where N is the number of records being fetched. This can lead to performance issues as the application has to make multiple database queries instead of fetching the required data in a more efficient manner.
- [ ] In the index action of the `PostsController`, the code is fetching the user's posts and eagerly loads the associated `author` for each post. However, the `comments` for each post are not being eagerly loaded, which means that for each post, an additional query will be executed to fetch its comments. To address the N+1 problem, you should also eagerly load the comments along with the posts for a user :+1: 
- Your code can look something like this 
```ruby
def index
  @user = User.find_by_id(params[:user_id])
  @posts = @user.posts.includes(:comments)
end
```
- [ ] Please don't forget to ensure that the `show` action also makes use of eager loading to load comments for a specific post :wink: 
- Your `show action` can look something like this 
```ruby
def show
  @current_user = current_user
  @user = User.find_by_id(params[:user_id])
  @post = Post.includes(:comments).find_by_id(params[:id])
  @comments = @post.comments
end
```
- [ ] Please include screenshots In your PR description of your app's logs before and after the fix :+1: